workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectCreationSystemHelper.cs

```diff
@@ -1,20 +1,26 @@
 using Improbable.Gdk.Core;
 using Unity.Entities;
+using UnityEngine;
 
 namespace Improbable.Gdk.GameObjectCreation
 {
     public static class GameObjectCreationSystemHelper
     {
-        public static void EnableStandardGameObjectCreation(World world)
+        public static void EnableStandardGameObjectCreation(World world, IEntityGameObjectCreator creator = null,
+            GameObject workerGameObject = null)
         {
-            var workerSystem = world.GetOrCreateManager<WorkerSystem>();
-            EnableStandardGameObjectCreation(world, new GameObjectCreatorFromMetadata(workerSystem.WorkerType,
-                workerSystem.Origin, workerSystem.LogDispatcher));
-        }
+            if (creator == null)
+            {
+                var workerSystem = world.GetOrCreateManager<WorkerSystem>();
+                creator = new GameObjectCreatorFromMetadata(workerSystem.WorkerType, workerSystem.Origin,
+                    workerSystem.LogDispatcher);
+            }
 
-        public static void EnableStandardGameObjectCreation(World world, IEntityGameObjectCreator creator)
-        {
             world.CreateManager<GameObjectInitializationSystem>(creator);
+            if (workerGameObject != null)
+            {
+                world.CreateManager<WorkerEntityGameObjectLinkerSystem>(workerGameObject);
```

Comment:
World gives you an error. I can't remember what but it won't add two systems.

Reply:
I guess what I tried to say is that it is not apparent right now when you should be allowed to call this method.
Call this method after you connected, and the worker gameobject is not set up correctly since you don't have the `OnConnected` tag anymore. Call this method a second time and it sometimes errors and sometimes doesn't depending on whether `workerGameObject` is null. It is also not apparent whether we allow the worker gameobject can be re-set to a different gameobject mid lifecycle.

Reply:
I assume that right now, the idea is to always call this method at most once ever and do that before we connect. If that's the case, we should intercept the cases where the method is called when it shouldn't be called and log a descriptive error message.